fieldpath/pathelementmap_test.go

```diff
@@ -29,22 +29,22 @@ func TestPathElementValueMap(t *testing.T) {
 		t.Fatal("Unexpected path-element found in empty map")
 	}
 
-	m.Insert(PathElement{FieldName: strptr("carrot")}, value.StringValue("knife"))
-	m.Insert(PathElement{FieldName: strptr("chive")}, value.IntValue(2))
+	m.Insert(PathElement{FieldName: strptr("carrot")}, value.ValueInterface{Value: "knife"})
+	m.Insert(PathElement{FieldName: strptr("chive")}, value.ValueInterface{Value: 2})
 
 	if _, ok := m.Get(PathElement{FieldName: strptr("onion")}); ok {
 		t.Fatal("Unexpected path-element in map")
 	}
 
 	if val, ok := m.Get(PathElement{FieldName: strptr("carrot")}); !ok {
 		t.Fatal("Missing path-element in map")
-	} else if !val.Equals(value.StringValue("knife")) {
+	} else if !value.Equals(val, value.ValueInterface{Value: "knife"}) {
```

Comment:
hm, I expected the interface to let you also not change this line.

Reply:
Yeah, I'm not sure yet ...